otel-integration/k8s-helm/e2e-test/main_test.go

```diff
@@ -142,12 +179,88 @@ func checkResourceAttributes(t *testing.T, attributes pcommon.Map, scopeName str
 
 	attributes.Range(func(k string, v pcommon.Value) bool {
 		val, ok := compareMap[k]
-		require.True(t, ok, "unexpected attribute %v - scopeName: %s", k, scopeName)
+		require.True(t, ok, "metrics: unexpected attribute %v - scopeName: %s", k, scopeName)
 		if val != "" {
-			require.Equal(t, val, v.AsString(), "unexpected value for attribute %v", k)
+			require.Equal(t, val, v.AsString(), "metrics: unexpected value for attribute %v", k)
 		}
 		return true
 	})
 
 	return nil
 }
+
+func checkTracesAttributes(t *testing.T, actual []ptrace.Traces, testID string, testNs string) error {
+	if len(actual) == 0 {
+		t.Fatal("No traces received")
+	}
+
+	for _, current := range actual {
+		trace := ptrace.NewTraces()
+		current.CopyTo(trace)
+
+		for i := 0; i < trace.ResourceSpans().Len(); i++ {
+			rspans := trace.ResourceSpans().At(i)
+
+			_, ok := expectedTracesSchemaURL[rspans.SchemaUrl()]
+			require.True(t, ok, "traces: resource %v does not match one of the expected values", rspans.SchemaUrl())
+			if ok {
+				expectedTracesSchemaURL[rspans.SchemaUrl()] = true
+			}
+
+			resource := trace.ResourceSpans().At(i).Resource()
+			service, exist := resource.Attributes().Get(ServiceNameAttribute)
+
+			expectedTrace := expectedTraces(testID, testNs)[service.AsString()]
+			require.NotEmpty(t, expectedTrace, "traces: unexpected service name %v", service.AsString())
+			require.True(t, exist, "traces: resource does not have the 'service.name' attribute")
+			assert.NoError(t, assertExpectedAttributes(resource.Attributes(), expectedTrace.attrs))
+
+			require.NotZero(t, trace.ResourceSpans().At(i).ScopeSpans().Len())
+			require.NotZero(t, trace.ResourceSpans().At(i).ScopeSpans().At(0).Spans().Len())
```

Comment:
Could use reuse `rspans` here.

```suggestion
			require.NotZero(t, rspans.At(i).ScopeSpans().Len())
			require.NotZero(t, rspans.At(i).ScopeSpans().At(0).Spans().Len())
```